src/app/hooks/queries/useAccountBalance.ts

```diff
@@ -0,0 +1,105 @@
+import useBtcClient from '@hooks/useBtcClient';
+import useNetworkSelector from '@hooks/useNetwork';
+import useWalletSelector from '@hooks/useWalletSelector';
+import {
+  Account,
+  BtcAddressData,
+  StxAddressData,
+  fetchStxAddressData,
+  microstacksToStx,
+  satsToBtc,
+} from '@secretkeylabs/xverse-core';
+import { setAccountBalanceAction } from '@stores/wallet/actions/actionCreators';
+import { PAGINATION_LIMIT } from '@utils/constants';
+import BigNumber from 'bignumber.js';
+import { useEffect, useRef, useState } from 'react';
+import { useDispatch } from 'react-redux';
+
+const useAccountBalance = () => {
+  const btcClient = useBtcClient();
+  const stacksNetworkInstance = useNetworkSelector();
+  const { btcFiatRate, stxBtcRate } = useWalletSelector();
+  const dispatch = useDispatch();
+  const queue = useRef<Account[]>([]);
+  const [isProcessingQueue, setIsProcessingQueue] = useState(false);
+  const [queueLength, setQueueLength] = useState(0);
+
+  const calculateTotalBalance = (btcBalance?: number, stxBalance?: number): string => {
+    let totalBalance = BigNumber(0);
+
+    if (btcBalance) {
+      const btcFiatEquiv = satsToBtc(BigNumber(btcBalance)).multipliedBy(BigNumber(btcFiatRate));
+      totalBalance = totalBalance.plus(btcFiatEquiv);
+    }
+
+    if (stxBalance) {
+      const stxFiatEquiv = microstacksToStx(BigNumber(stxBalance))
+        .multipliedBy(BigNumber(stxBtcRate))
+        .multipliedBy(BigNumber(btcFiatRate));
+      totalBalance = totalBalance.plus(stxFiatEquiv);
+    }
+
+    return totalBalance.toNumber().toFixed(2);
+  };
+
+  const fetchBalances = async (account: Account | null) => {
+    if (!account) {
+      return;
+    }
+
+    let btcBalance = 0;
+    let stxBalance = 0;
+
+    if (account.btcAddress) {
+      const btcData: BtcAddressData = await btcClient.getBalance(account?.btcAddress);
+      btcBalance = btcData.finalBalance;
+    }
+
+    if (account.stxAddress) {
+      const stxData: StxAddressData = await fetchStxAddressData(
```

Comment:
i think we can improve timing and resources here by avoiding the use of this method and calling directly to 

``` ts
const apiUrl = `${getNetworkURL(network)}/v2/accounts/${stxAddress}?proof=0`;
```

cause fetchStxAddressData also makes calls to tx, pending tx and process them. and we only care about balance here right?

Reply:
yeah, that definitely makes sense 👍